### PR TITLE
[CALCITE-3539] Implement INSTR function

### PR DESCRIPTION
**Pull request by xy2953396112**

### CODE REVIEW

**Comment by jinxing64** [comment]
Do we also need to update reference.md

**Comment by xy2953396112** [comment]
> Do we also need to update reference.md ?

Thansk, i have updated reference.md.

File: core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java [hunk 1]
```diff
@@ -4508,6 +4508,15 @@ private void checkNullOperand(SqlTester tester, String op) {
     tester1.checkNull("STRCMP(cast(null as varchar(1)), 'mytesttext')");
   }
 
+  @Test void testInstrFunc() {
+    final SqlTester tester1 = tester(SqlLibrary.MYSQL);
+    tester1.setFor(SqlLibraryOperators.INSTR);
+    tester1.checkString("INSTR('foobarbar', 'bar')", "4", "INTEGER NOT NULL");
+    tester1.checkString("INSTR('bar', 'foobarbar')", "0", "INTEGER NOT NULL");
+    tester1.checkNull("INSTR('foobarbar', cast(null as varchar(1)))");
+    tester1.checkNull("INSTR(cast(null as varchar(1)), 'foobarbar')");
```
Comment (chunweilei) [inline core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java:4517]: Could you also add negative tests?

File: site/_docs/reference.md [hunk 2]
```diff
@@ -2359,6 +2359,7 @@ semantics.
 | m | SPACE(integer)                                 | Returns a string of *integer* spaces; returns an empty string if *integer* is less than 1
 | o | SUBSTR(string, position [, substringLength ]) | Returns a portion of *string*, beginning at character *position*, *substringLength* characters long. SUBSTR calculates lengths using characters as defined by the input character set
 | m | STRCMP(string, string)                         | Returns 0 if both of the strings are same and returns -1 when the first argument is smaller than the second and 1 when the second one is smaller the first one.
+| m | INSTR(string, string)                          | Returns the index of the first occurrence of the second argument in the first argument and returns 0 if the first argument not include the second argument.
```
Comment (DonnyZone) [inline site/_docs/reference.md:2362]: We'd better remove the period at the end of the description.

**Comment by DonnyZone** [comment]
MySQL's document says: '`This function is multibyte safe, and is case-sensitive only if at least one argument is a binary string.`' 
It is better to add several tests for case-(in)sensitive.

File: site/_docs/reference.md (hunk 2 above)
Comment (XuQianJin-Stars) [inline site/_docs/reference.md:2362]: Please move the description here and sort by alphabet.

**Comment by XuQianJin-Stars** [comment]
hi @xy2953396112  There is an error in the title here, it should be the link https://issues.apache.org/jira/browse/CALCITE-3959

File: site/_docs/reference.md (hunk 2 above)
Comment (zinking) [reply site/_docs/reference.md]: well, STRCMP does the same, change it altogether

**Comment by zinking** [comment]
looks good overall

**Pull request closed without merging**